cmd/tailscale/cli: TestUpdatePrefs/advertise_connector fails on macOS

At head (f13b2bce93b9bd), this test fails on macOS, but not Linux:

```
$ ./tool/go test ./cmd/tailscale/cli
--- FAIL: TestUpdatePrefs (0.00s)
    --- FAIL: TestUpdatePrefs/advertise_connector (0.00s)
        cli_test.go:1352: aborted, no changes made
FAIL
FAIL	tailscale.com/cmd/tailscale/cli	0.318s
FAIL
```

/cc @andrew-d

--- a/cmd/tailscale/cli/up.go
+++ b/cmd/tailscale/cli/up.go
@@ -379,7 +379,7 @@ func updatePrefs(prefs, curPrefs *ipn.Prefs, env upCheckEnv) (simpleUp bool, jus
 		return false, nil, err
 	}
 
-	if runtime.GOOS == "darwin" && env.upArgs.advertiseConnector {
+	if env.goos == "darwin" && env.upArgs.advertiseConnector {
 		if err := presentRiskToUser(riskMacAppConnector, riskMacAppConnectorMessage, env.upArgs.acceptedRisks); err != nil {
 			return false, nil, err
 		}

